test/iron-input.html

```diff
@@ -138,6 +163,27 @@
         assert.equal(input.bindValue, '123');
       });
 
+      test('prevent invalid input works in _onInput and does not use pattern', function() {
```

Comment:
I think "prevent-invalid-input works" is a good enough test name (the test isn't using `pattern` anywhere, so the test name is confusing outside of this PR)